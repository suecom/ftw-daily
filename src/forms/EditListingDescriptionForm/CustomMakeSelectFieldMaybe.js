import React from 'react';
import { required } from '../../util/validators';
import { FieldSelect } from '../../components';

import css from './EditListingDescriptionForm.css';

const CustomMakeSelectFieldMaybe = props => {
  const { name, id, make, intl } = props;
  const makeLabel = intl.formatMessage({
    id: 'EditListingDescriptionForm.makeLabel',
  });
  const makePlaceholder = intl.formatMessage({
    id: 'EditListingDescriptionForm.makePlaceholder',
  });
  const makeRequired = required(
    intl.formatMessage({
      id: 'EditListingDescriptionForm.makeRequired',
    })
  );
  return make ? (
    <FieldSelect className={css.make} name={name} id={id} label={makeLabel} validate={makeRequired}>
      <option disabled value="">
        {makePlaceholder}
      </option>
      {make.map(c => (
        <option key={c.key} value={c.key}>
          {c.label}
        </option>
      ))}
    </FieldSelect>
  ) : null;
};

export default CustomMakeSelectFieldMaybe;
