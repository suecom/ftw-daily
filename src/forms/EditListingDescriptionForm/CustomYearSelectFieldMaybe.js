import React from 'react';
import { required } from '../../util/validators';
import { FieldSelect } from '../../components';

import css from './EditListingDescriptionForm.css';

const CustomYearSelectFieldMaybe = props => {
  const { name, id, year, intl } = props;
  const yearLabel = intl.formatMessage({
    id: 'EditListingDescriptionForm.yearLabel',
  });
  const yearPlaceholder = intl.formatMessage({
    id: 'EditListingDescriptionForm.yearPlaceholder',
  });
  const yearRequired = required(
    intl.formatMessage({
      id: 'EditListingDescriptionForm.yearRequired',
    })
  );
  return year ? (
    <FieldSelect
      className={css.year}
      name={name}
      id={id}
      label={yearLabel}
      validate={yearRequired}
    >
      <option disabled value="">
        {yearPlaceholder}
      </option>
      {year.map(c => (
        <option key={c.key} value={c.key}>
          {c.label}
        </option>
      ))}
    </FieldSelect>
  ) : null;
};

export default CustomYearSelectFieldMaybe;
