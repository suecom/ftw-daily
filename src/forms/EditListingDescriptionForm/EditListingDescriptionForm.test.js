import React from 'react';
import { renderDeep } from '../../util/test-helpers';
import { fakeIntl } from '../../util/test-data';
import EditListingDescriptionForm from './EditListingDescriptionForm';

const noop = () => null;

describe('EditListingDescriptionForm', () => {
  it('matches snapshot', () => {
    const tree = renderDeep(
      <EditListingDescriptionForm
        intl={fakeIntl}
        dispatch={noop}
        onSubmit={v => v}
        saveActionMsg="Save description"
        updated={false}
        updateInProgress={false}
        make={[{ key: 'audi', label: 'Audi' }, { key: 'alvis', label: 'Alvis' }]}
        year={[{ key: 2004, label: 2004 }, { key: 2003, label: 2003 }]}
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
