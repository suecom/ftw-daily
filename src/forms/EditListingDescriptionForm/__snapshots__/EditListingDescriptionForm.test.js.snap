// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditListingDescriptionForm matches snapshot 1`] = `
<form
  action="/"
  className="root"
  method="post"
  onSubmit={[Function]}
>
  <div
    className="root title"
  >
    <label
      htmlFor="title"
    >
      EditListingDescriptionForm.title
    </label>
    <input
      autoFocus={true}
      className="input"
      id="title"
      maxLength={60}
      name="title"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="EditListingDescriptionForm.titlePlaceholder"
      type="text"
      value=""
    />
  </div>
  <div
    className="root description"
  >
    <label
      htmlFor="make"
    >
      EditListingDescriptionForm.makeLabel
    </label>
    <select
      className=""
      id="make"
      name="make"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      value=""
    >
      <option
        disabled={true}
        value=""
      >
        EditListingDescriptionForm.makePlaceholder
      </option>
      <option
        key="audi"
        value="audi"
      >
        Audi
      </option>
      <option
        key="alvis"
        value="alvis"
      >
        Alvis
      </option>
    </select>
  </div>
  <div
    className=""
  >
    <label
      htmlFor="model"
    >
      EditListingDescriptionForm.model
    </label>
    <input
      className=""
      id="model"
      name="model"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="EditListingDescriptionForm.modelPlaceholder"
      type="text"
      value=""
    />
  </div>
  <div
    className="root category"
  >
    <label
      htmlFor="year"
    >
      EditListingDescriptionForm.yearLabel
    </label>
    <select
      className=""
      id="year"
      name="year"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      value=""
    >
      <option
        disabled={true}
        value=""
      >
        EditListingDescriptionForm.yearPlaceholder
      </option>
      <option
        key="2004"
        value={2004}
      >
        2004
      </option>
      <option
        key="2003"
        value={2003}
      >
        2003
      </option>
    </select>
  </div>
  <div
    className=""
  >
    <label
      htmlFor="description"
    >
      EditListingDescriptionForm.description
    </label>
    <textarea
      className="undefined"
      id="description"
      maxLength={5000}
      name="description"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="EditListingDescriptionForm.descriptionPlaceholder"
      rows={1}
      value=""
    />
  </div>
  <button
    className="root submitButton"
    disabled={true}
    type="submit"
  >
    Save description
  </button>
</form>
`;
